pkg/datastore/test/caveat.go

```diff
@@ -1,65 +1,79 @@
-package memdb
+package test
 
 import (
 	"context"
+	"fmt"
 	"testing"
 	"time"
 
-	"github.com/google/go-cmp/cmp"
-	"google.golang.org/protobuf/testing/protocmp"
-
 	"github.com/authzed/spicedb/internal/datastore/common"
 	"github.com/authzed/spicedb/internal/testfixtures"
 	"github.com/authzed/spicedb/pkg/caveats"
 	"github.com/authzed/spicedb/pkg/datastore"
 	core "github.com/authzed/spicedb/pkg/proto/core/v1"
 	"github.com/authzed/spicedb/pkg/tuple"
 
+	"github.com/google/go-cmp/cmp"
 	"github.com/google/uuid"
 	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/testing/protocmp"
 	"google.golang.org/protobuf/types/known/structpb"
 )
 
-func TestWriteReadCaveat(t *testing.T) {
+func WriteReadDeleteCaveatTest(t *testing.T, tester DatastoreTester) {
 	req := require.New(t)
-
-	ds, err := NewMemdbDatastore(0, 1*time.Hour, 1*time.Hour)
+	ds, err := tester.New(0*time.Second, veryLargeGCWindow, 1)
 	req.NoError(err)
 
-	// Dupes in same transaction are treated as upserts
-	coreCaveat := createCoreCaveat(t)
+	skipIfNotCaveatStorer(t, ds)
+
+	// Dupes in same transaction are fail to be written
 	ctx := context.Background()
+	coreCaveat := createCoreCaveat(t)
 	_, err = writeCaveats(ctx, ds, coreCaveat, coreCaveat)
```

Comment:
Write some non-empty context here and verify on the read that it is the same map, with the same values

Reply:
There is a test for this already: `WriteCaveatedRelationshipTest`

Reply:
Please add a table driven test with different context maps\

Reply:
thoughts on doing it in my follow up?